cpp/unreal_plugins/SimulationController/Source/SimulationController/UrdfRobotAgent.cpp

```diff
@@ -0,0 +1,175 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "SimulationController/UrdfRobotAgent.h"
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <Camera/CameraComponent.h>
+#include <Components/SceneCaptureComponent2D.h>
+#include <Engine/World.h>
+#include <Math/Rotator.h>
+#include <Math/Vector.h>
+
+#include "CoreUtils/ArrayDesc.h"
+#include "CoreUtils/Assert.h"
+#include "CoreUtils/Config.h"
+#include "CoreUtils/Log.h"
+#include "CoreUtils/Std.h"
+#include "CoreUtils/Unreal.h"
+#include "SimulationController/CameraSensor.h"
+#include "UrdfRobot/UrdfLinkComponent.h"
+#include "UrdfRobot/UrdfRobotComponent.h"
+#include "UrdfRobot/UrdfRobotPawn.h"
+
+UrdfRobotAgent::UrdfRobotAgent(UWorld* world)
+{
+    SP_LOG_CURRENT_FUNCTION();
+
+    FVector spawn_location = FVector::ZeroVector;
+    FRotator spawn_rotation = FRotator::ZeroRotator;
+    std::string spawn_mode = Config::get<std::string>("SIMULATION_CONTROLLER.URDF_ROBOT_AGENT.SPAWN_MODE");
+    if (spawn_mode == "specify_existing_actor") {
+        AActor* spawn_actor = Unreal::findActorByName(world, Config::get<std::string>("SIMULATION_CONTROLLER.URDF_ROBOT_AGENT.SPAWN_ACTOR_NAME"));
+        SP_ASSERT(spawn_actor);
+        spawn_location = spawn_actor->GetActorLocation();
+        spawn_rotation = spawn_actor->GetActorRotation();
+    } else if (spawn_mode == "specify_pose") {
+        spawn_location = FVector(
+            Config::get<float>("SIMULATION_CONTROLLER.URDF_ROBOT_AGENT.SPAWN_LOCATION_X"),
+            Config::get<float>("SIMULATION_CONTROLLER.URDF_ROBOT_AGENT.SPAWN_LOCATION_Y"),
+            Config::get<float>("SIMULATION_CONTROLLER.URDF_ROBOT_AGENT.SPAWN_LOCATION_Z"));
+        spawn_rotation = FRotator(
+            Config::get<float>("SIMULATION_CONTROLLER.URDF_ROBOT_AGENT.SPAWN_ROTATION_PITCH"),
+            Config::get<float>("SIMULATION_CONTROLLER.URDF_ROBOT_AGENT.SPAWN_ROTATION_YAW"),
+            Config::get<float>("SIMULATION_CONTROLLER.URDF_ROBOT_AGENT.SPAWN_ROTATION_ROLL"));
+    } else {
+        SP_ASSERT(false);
+    }
+    FActorSpawnParameters actor_spawn_params;
+    actor_spawn_params.Name = Unreal::toFName(Config::get<std::string>("SIMULATION_CONTROLLER.URDF_ROBOT_AGENT.URDF_ROBOT_ACTOR_NAME"));
+    actor_spawn_params.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+    urdf_robot_pawn_ = world->SpawnActor<AUrdfRobotPawn>(spawn_location, spawn_rotation, actor_spawn_params);
+    SP_ASSERT(urdf_robot_pawn_);
+
+    // Initialize the urdf_robot_pawn
+    urdf_robot_pawn_->Initialize();
```

Comment:
I think we need to specify which URDF file gets loaded before calling `Initialize`.

Reply:
The URDF file is determined during the `Initialize()` call in `UrdfRobotPawn.cpp` file (https://github.com/isl-org/spear/blob/b2c8439faf0729b6d3db3768f4f0e78d8612ea7f/cpp/unreal_plugins/UrdfRobot/Source/UrdfRobot/UrdfRobotPawn.cpp#L56). The current behavior is that if the config system is initialized which is the case while we are using a `UrdfRobotAgent`, the URDF file specified as part of our config system is used. I think this behavior makes sense. 

Are you suggesting we change the order of the if statements above in `UrdfRobotPawn.cpp` to give precedence to `UrdfFile` being not-empty over the check to see if the config system is initialized?

Reply:
Ah, my bad. I think the current behavior is what we want, and this aspect of `UrdfRobotAgent` is implemented correctly. Disregard my comment here.